### PR TITLE
Low-level controller 

### PR DESCRIPTION
**Pull request by PerFrivik**
Closed #53 and re-opened it to the correct branch! 

Thank you for the reminder @felix-ch

### CODE REVIEW

File: smb_control/src/PIDControllerNode.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+    // Publish the control output
+    cmd_vel_pub_.publish(cmd_vel_);
+}
+
+void PIDControllerNode::checkTimeout()
+{
+
+    if (ros::Time::now() - last_twist_time_ > ros::Duration(odom_timeout_)) {
+        // No new message within 0.2 seconds, setting cmd_vel_requested_ to zero
```
Comment (mantelt) [inline smb_control/src/PIDControllerNode.cpp:198]: comment is misleading. timeout is a parameter, not fixed

File: smb_control/src/PIDControllerNode.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+}
+
+PIDControllerNode::~PIDControllerNode()
+{
+    // Cleanup can be handled here
+}
+
+void PIDControllerNode::run()
+{
+    ros::Rate loop_rate(100);  // 10 Hz
```
Comment (mantelt) [inline smb_control/src/PIDControllerNode.cpp:33]: comment is misleading

**Review by mantelt — CHANGES_REQUESTED** [review]
nice work, don't see any real issues. just a few really minor things,

File: smb_control/launch/smb_control.launch [hunk 3]
```diff
@@ -51,16 +50,17 @@
...
+
 
   <!-- Start EKF for localization -->
-  <group if="$(arg enable_ekf)" >
-    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" ns="$(arg control_namespace)">
-      <rosparam command="load" file="$(find smb_control)/config/localization.yaml" />
-    </node>
-  </group>
+
+  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" output="screen">
```
[the remark this answers is not in the document]
Comment (PerFrivik) [reply smb_control/launch/smb_control.launch:62]: The EKF topic is: /filtered/odometry 

Should I add a comment for that?

File: smb_control/src/PIDControllerNode.cpp (hunk 2 above)
Comment (PerFrivik) [reply smb_control/src/PIDControllerNode.cpp]: Fixed and changed 100 Hz!

File: smb_control/src/PIDControllerNode.cpp (hunk 1 above)
Comment (PerFrivik) [reply smb_control/src/PIDControllerNode.cpp]: Removed the comment, currently its fixly defined as 0.1, I could add it as a parameter if you want? but I think this is quite dangerous to play with. I just added a member variable to make more clear, but its not really nessesary and a good design choice.

**Review by mantelt — APPROVED** [review]

**Pull request merged**